### PR TITLE
User can create a post

### PR DESCRIPTION
**Pull request by rithreaksa**
[User can create a post with a song attached](https://www.pivotaltracker.com/story/show/176623786)
### User Story
```
As a an API,            
In order to allow the client to share content to make a post,         
I want to be able to include search result in a post.
```
## Changes proposed in this pull request:
* Adding a route for creating a POST
* Implementing create action
* Adding validation test for post model in spec

## What we have learned working on this feature:
* The importance of naming convention
* How to implement dynamic error message

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -16,10 +16,10 @@
...
   create_table "posts", force: :cascade do |t|
-    t.string "track_name"
+    t.string "track"
     t.string "artists"
     t.text "image"
-    t.text "preview_url"
+    t.text "preview"
     t.text "description"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
```
Comment (sealfury) [inline db/schema.rb:25]: thanks for making the changes with the naming conventions!

File: app/controllers/api/posts_controller.rb [hunk 2]
```diff
@@ -3,4 +3,19 @@ def index
     posts = Post.all
     render json: { posts: posts }
   end
+
+  def create
+    post = Post.create(post_params)
+    if post.persisted?
+      render json: { message: 'Your post was successfully created!' }, status: 201
```
Comment (ViktorHek) [inline app/controllers/api/posts_controller.rb:10]: maybe change this message to something fun. we are probably not going to use it in the front end so go nuts!

File: db/migrate/20210125135243_create_posts.rb [hunk 3]
```diff
@@ -1,10 +1,10 @@
 class CreatePosts < ActiveRecord::Migration[6.0]
   def change
     create_table :posts do |t|
-      t.string :track_name
+      t.string :track
       t.string :artists
       t.text :image
-      t.text :preview_url
+      t.text :preview
```
Comment (ViktorHek) [inline db/migrate/20210125135243_create_posts.rb:7]: nice!

File: app/models/post.rb [hunk 4]
```diff
@@ -1,2 +1,3 @@
 class Post < ApplicationRecord
+  validates_presence_of :track, :artists, :image, :preview, :description
```
Comment (ViktorHek) [inline app/models/post.rb:2]: maybe we users don't NEED to have a description??

File: spec/requests/api/user_can_create_a_post_spec.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
+RSpec.describe 'POST /api/posts', type: :request do
+  describe 'successfully create a post' do
+    before do
+      post '/api/posts',
+           params: {
+             post: {
+               track: 'One Beer',
+               artists: 'MF DOOM',
```
Comment (ViktorHek) [inline spec/requests/api/user_can_create_a_post_spec.rb:8]: change the song to "CELLZ" just for the fun of it

File: app/controllers/api/posts_controller.rb (hunk 2 above)
Comment (rithreaksa) [reply app/controllers/api/posts_controller.rb:10]: @ViktorHek  We'll take this into consideration!

File: app/models/post.rb (hunk 4 above)
Comment (salindholm) [reply app/models/post.rb:2]: We agreed in the team that the purpose of the app is to interact through music, so we will keep the description as a validation!

File: spec/requests/api/user_can_create_a_post_spec.rb (hunk 5 above)
Comment (salindholm) [reply spec/requests/api/user_can_create_a_post_spec.rb:8]: We will go for this song for now, we can add the other one next time!

**Pull request merged**